jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/list/ProcessInstanceListPresenter.java

```diff
@@ -164,8 +164,7 @@ protected DataSetReadyCallback getDataSetReadyCallback(final Integer startRange,
                             lastPage = true;
                         }
 
-                        final String filterValue = isFilteredByProcessId(tableSettings.getDataSetLookup().getOperationList());
-                        if (filterValue != null) {
+                        if (preferences.getLoadProcessVariablesByDefault() || isFilteredByProcessId(tableSettings.getDataSetLookup().getOperationList()) != null) {
```

Comment:
@cristianonicolai In this case:
- The preference is set to loadProcessVariablesByDefault.
- The user selects one of the domain columns and the table shows the right value
- The preference is changed to Not loadProcessVariablesByDefault.
- When the user access to the processInstances again, This domain column is still present but this condition makes that the content is not loaded. It can be seen a bit inconsistent regarding it don't show the value because of the preferences, maybe we have to check if the displayed columns are available in the applied dataset. What do you think?

Reply:
@nmirasch yeah, I saw that, but not sure how to get around, because in this case we're mixing variables from different tasks and processes, so how to differentiate that these are empty or the preference has been turned of?

Reply:
@cristianonicolai maybe create a new method at ProcessInstanceListViewImpl and remove that extra columns if that condition at the presenter don't retrieve the dataset that could fill the content..something like this:
    public void removeDomainSpecifColumns(ExtendedPagedTable<ProcessInstanceSummary> extendedPagedTable) {
        for (ColumnMeta<ProcessInstanceSummary> cm : extendedPagedTable.getColumnMetaList()) {
            if (cm.isExtraColumn()) {
                extendedPagedTable.removeColumnMeta(cm);
            }
        }
        extendedPagedTable.storeColumnToPreferences();
    }
It's true that the extra column selection is lost, but at least it prevents empty content when it's not empty. It's just an idea but we have to think if to add this call makes sense regarding the possible situation it tries to prevent.